skimage/transform/steerable.py

```diff
@@ -84,6 +84,13 @@ def __init__(self, height=5, nbands=4):
         nbands : int
             number of orientations in Steerable decomposition
         """
+
+        if not (isinstance(height, int) and isinstance(nbands, int)):
```

Comment:
isinstance with `int` is too restrictive  (`isinstance(np.int32(5), int)` returns False on most machines)

`isinstance(height, np.integer)` would match all integer types, but I tend to favor the following form as it allows floats with integer value as well:
`(height % 1 == 0)`

If you go the later route, you probably want to set, `self.height = int(height)` below so integer-like floats get converted to actual integers.